src/controls/Control.ts

```diff
@@ -332,33 +342,62 @@ export class Control {
     };
   }
 
+  renderConnection(
+    ctx: CanvasRenderingContext2D,
+    from: Point,
+    to: Point,
+    styleOverride: Pick<
+      ControlRenderingStyleOverride,
+      'borderColor' | 'borderDashArray'
+    > = {},
+    fabricObject: FabricObject
+  ) {
+    ctx.save();
+    ctx.strokeStyle = styleOverride.borderColor || fabricObject.borderColor;
+    fabricObject._setLineDash(
+      ctx,
+      styleOverride.borderDashArray || fabricObject.borderDashArray
+    );
+    ctx.beginPath();
+    ctx.moveTo(from.x, from.y);
+    ctx.lineTo(to.x, to.y);
+    ctx.closePath();
+    ctx.stroke();
+    ctx.restore();
+  }
+
   /**
    * Render function for the control.
    * When this function runs the context is unscaled. unrotate. Just retina scaled.
    * all the functions will have to translate to the point left,top before starting Drawing
    * if they want to draw a control where the position is detected.
    * left and top are the result of the positionHandler function
    * @param {RenderingContext2D} ctx the context where the control will be drawn
-   * @param {Number} left position of the canvas where we are about to render the control.
-   * @param {Number} top position of the canvas where we are about to render the control.
+   * @param {Point} position coordinate where the control center should be
    * @param {Object} styleOverride
    * @param {FabricObject} fabricObject the object where the control is about to be rendered
    */
   render(
     ctx: CanvasRenderingContext2D,
-    left: number,
-    top: number,
-    styleOverride: ControlRenderingStyleOverride | undefined,
+    position: TOCoord,
+    styleOverride: ControlRenderingStyleOverride = {},
     fabricObject: FabricObject
   ) {
-    styleOverride = styleOverride || {};
+    if (this.withConnection) {
+      this.renderConnection(
+        ctx,
+        position.connection,
```

Comment:
oh i see this is the breaking issue.
You can just call this separately no?

Reply:
correct